code/modules/medals/rewardsLocker.dm

```diff
@@ -850,6 +850,18 @@
 					H.set_clothing_icon_dirty()
 					succ = TRUE
 
+			if (H.gloves)
+				var/obj/item/clothing/gloves/M = H.gloves
+				if (istype(M, /obj/item/clothing/gloves/swat/captain))
+					var/prev = M.name
+					M.icon_state = "centcomgloves"
+					M.item_state = "centcomgloves"
+					M.name = "commander's gloves"
+					M.real_name = "commander's gloves"
+					M.desc = "A pair of formal gloves that are electrically insulated and quite heat-resistant. (Base Item: [prev])"
+					H.set_clothing_icon_dirty()
```

Comment:
I think you can just `H.update_gloves(H.mutantrace.hand_offset)` here. Clothing icon dirty updates every piece of clothing + damage + blood + name + etc (god why, that isn't clothing) which isn't necessary here.

Reply:
could do update_clothing at the end instead of per item? maybe not if there's a chance of a runtime between?